### ISSUE TITLE
Make sstable interoperable between machine of different endianess

### ISSUE THREAD

**Issue by raphaelsc** [scylladb/scylla#3056]
Some fields of summary file are written using native machine order, which makes sstable not portable between machine of different endianess. C* probably designed it that way to make their life easier with memory mapping. @glommer suggested that we could make little endian the default byte ordering for those fields, such that a sstable generated on PowerPC's Scylla could be operated both on X86's Scylla and C*. If a sstable generated on PowerPC's C* is moved to x86's ScyllaDB, we could detect it (offset for summary entry will be really high, higher than the summary file size) and rebuild the summary like we do when it's missing.

**Comment by slivne** [comment]
@raphaelsc / @vladzcloudius how do we support this in the tools ?

It would be better if the structure on the disk will be consistent in my view - e.g. all use the x86 structure - whats the penalty in this case ?

**Comment by vladzcloudius** [comment]
@slivne @raphaelsc I agree. We'd better choose some endianess and use it on all platforms.

**Comment by raphaelsc** [comment]
@slivne @vladzcloudius I think after 331b906d9315b303875672d7f81c2b95633c3c54, we're now consistent. For everything else, network order is used, but summary entries for some reason (decided in C* not us) that host endianess would be used. Perhaps they load summary on demand which would explain the decision.

Now, sstable generated on a big endian host can now be loaded on little endian one, and vice-versa. But a big endian sstable generated by C* cannot be loaded by a little endian host because they don't follow the aforementioned patch.

**Comment by vladzcloudius** [comment]
@slivne @raphaelsc @glommer @avikivity I think that the most important transition path that we need to support is from C* to Scylla on the same platform, e.g. transition the existing C* cluster on Power configured in a big endian mode to Scylla. The patch above seem to break this path.

However, note that scylla doesn't support any big endian platform yet (nor it has been attempted to be compiled/executed on any).

**Comment by vladzcloudius** [comment]
@glommer I think, as long as scylla doesn't support a BIG ENDIAN platform, this issue is not a bug. We should keep this in mind however I don't think that a transition of C* cluster built on a BE machine to scylla on a LE in unreasonable since the same transition won't work with C*. If this kind of a transition is required anyway we should suggest a customer to use a different path of data migration (as painful as it would). 

And in all other cases we should be more that ok with the current state of our code.
